### PR TITLE
Add --rasa-x-port

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
- add argument `--rasa-x-port` for `rasa x` command
- print error if port of rasa x and rasa server are the same

related to https://github.com/RasaHQ/rasa/issues/3597

depends on https://github.com/RasaHQ/rasa-x/pull/776/files

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/cli/x.py [hunk 1]
```diff
@@ -220,67 +226,38 @@ def is_rasa_project_setup(project_path: Text):
...
 
 
-def rasa_x(args: argparse.Namespace):
-    from rasa.cli.utils import print_success, print_error, signal_handler
-    from rasa.core.utils import AvailableEndpoints
+def _validate_rasa_x_start(args: argparse.Namespace, project_path: Text):
+    if not is_rasa_x_installed():
+        print_error(
+            "Rasa X is not installed. The `rasa x` "
+            "command requires an installation of Rasa X."
```
Comment (tmbo) [inline rasa/cli/x.py]: while we are here: can we add the installation instructions here, or at least a link to the docs?

File: rasa/cli/x.py [hunk 2]
```diff
@@ -220,67 +226,38 @@ def is_rasa_project_setup(project_path: Text):
...
+        )
+        sys.exit(1)
 
-    signal.signal(signal.SIGINT, signal_handler)
+    if args.port == args.rasa_x_port:
+        print_error(
+            "The port for Rasa X and the port of the Rasa server are the same. "
```
Comment (tmbo) [inline rasa/cli/x.py]: let's also add the configured port to the error message.

File: rasa/cli/x.py [hunk 3]
```diff
@@ -220,67 +226,38 @@ def is_rasa_project_setup(project_path: Text):
...
+            "different ports using the arguments '--port' and '--rasa-x-port'."
+        )
+        sys.exit(1)
 
-    _configure_logging(args)
+    if not is_rasa_project_setup(project_path):
+        print_error(
+            "This directory is not a valid Rasa project. Use 'rasa init' "
+            "to create a new Rasa project or switch to a valid Rasa project "
+            "directory."
```
Comment (tmbo) [inline rasa/cli/x.py]: can we also add a link to some docs here? (I think there might be something about the directory structure)

File: CHANGELOG.rst [hunk 4]
```diff
@@ -11,6 +11,7 @@ This project adheres to `Semantic Versioning`_ starting with version 1.0.
 
 Added
 -----
+- added argument ``--rasa-x-port`` to specify the port of Rasa X when running Rasa X locally via ``rasa x``
```
Comment (tmbo) [inline CHANGELOG.rst]: should we also add this to the docs somewhere? (e.g. as a note)

Comment (tabergma) [reply CHANGELOG.rst]: We don't show any of the options you have for `rasa x`. What do you think, should we add them here: http://rasa.com/docs/rasa-x/installation-and-setup/? Similar to http://rasa.com/docs/rasa/user-guide/command-line-interface/. Or do you think we should just add the most important ones?

Comment (tabergma) [reply CHANGELOG.rst]: Added it to http://rasa.com/docs/rasa/user-guide/command-line-interface/

Comment (tmbo) [reply CHANGELOG.rst]: sounds great, thanks :+1:

**Pull request merged**